service/Checkout.php

```diff
@@ -76,11 +77,11 @@ public function requireFetchPaymentMethods(Cart $cart)
             }
 
             // Check if all keys exist, and then check if the delivery step is complete and payment method is reachable
-            if (array_key_exists(self::DELIVERY_STEP, $jsonData) &&
+            if (($isVirtualCart || array_key_exists(self::DELIVERY_STEP, $jsonData)) &&
                 array_key_exists(self::IS_COMPLETE, $jsonData[self::DELIVERY_STEP]) &&
```

Comment:
line 81 would break (undefined index) if `$isVirtualCart` is `true` and `self::DELIVERY_STEP` does not exist in the `$jsonData` array

Reply:
there should be a simpler way to do this using `isset()` (like `isset($jsonData[self::DELIVERY_STEP][self::IS_COMPLETE])`) or better still using null coalesce operator (if you only support PHP 7 and above)

Reply:
Updated and cleaned up the if condition a bit in https://github.com/Adyen/adyen-prestashop/pull/237/commits/ded6de69944df0d6948413cc161ced4685ed338c